components/script/dom/htmllinkelement.rs

```diff
@@ -273,14 +276,24 @@ impl PreInvoke for StylesheetContext {}
 impl AsyncResponseListener for StylesheetContext {
     fn headers_available(&mut self, metadata: Result<Metadata, NetworkError>) {
         self.metadata = metadata.ok();
+        if let Some(ref meta) = self.metadata {
+            if let Some(ContentType(Mime(TopLevel::Text, SubLevel::Css, _))) = meta.content_type {
+            } else {
+                self.elem.root().upcast::<EventTarget>().fire_simple_event("error");
+            }
+        }
     }
 
     fn data_available(&mut self, payload: Vec<u8>) {
         let mut payload = payload;
         self.data.append(&mut payload);
     }
 
-    fn response_complete(&mut self, _status: Result<(), NetworkError>) {
+    fn response_complete(&mut self, status: Result<(), NetworkError>) {
+        if status.is_err() {
```

Comment:
Why the NetworkError doesn't happen here right now?
After adding the `assert!(status.is_ok());` here, the assertion will failed when I try to disable the network connection, and open a local HTML file with remote CSS file.